samples/api/oit_depth_peeling/oit_depth_peeling.cpp

```diff
@@ -0,0 +1,558 @@
+		    vkb::initializers::descriptor_set_layout_binding(VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, VK_SHADER_STAGE_FRAGMENT_BIT, 2, kLayerMaxCount),
+		};
+		VkDescriptorSetLayoutCreateInfo descriptor_layout_create_info = vkb::initializers::descriptor_set_layout_create_info(set_layout_bindings.data(), static_cast<uint32_t>(set_layout_bindings.size()));
+		VK_CHECK(vkCreateDescriptorSetLayout(get_device().get_handle(), &descriptor_layout_create_info, nullptr, &combine_descriptor_set_layout));
+	}
+
+	{
+		std::vector<VkDescriptorPoolSize> pool_sizes = {
+		    vkb::initializers::descriptor_pool_size(VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 4),
+		    vkb::initializers::descriptor_pool_size(VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, 11),
+		};
+		const uint32_t             num_descriptor_sets         = 3;
+		VkDescriptorPoolCreateInfo descriptor_pool_create_info = vkb::initializers::descriptor_pool_create_info(static_cast<uint32_t>(pool_sizes.size()), pool_sizes.data(), num_descriptor_sets);
+		VK_CHECK(vkCreateDescriptorPool(get_device().get_handle(), &descriptor_pool_create_info, nullptr, &descriptor_pool));
+	}
+
+	{
+		const VkDescriptorSetLayout layouts[kDepthCount] = {gather_descriptor_set_layout, gather_descriptor_set_layout};
+		VkDescriptorSetAllocateInfo alloc_info           = vkb::initializers::descriptor_set_allocate_info(descriptor_pool, layouts, kDepthCount);
+		VK_CHECK(vkAllocateDescriptorSets(get_device().get_handle(), &alloc_info, gather_descriptor_set));
+	}
+
+	{
+		VkDescriptorSetAllocateInfo alloc_info = vkb::initializers::descriptor_set_allocate_info(descriptor_pool, &combine_descriptor_set_layout, 1);
+		VK_CHECK(vkAllocateDescriptorSets(get_device().get_handle(), &alloc_info, &combine_descriptor_set));
+	}
+}
+
+void OITDepthPeeling::update_descriptors()
+{
+	VkDescriptorBufferInfo scene_constants_descriptor = create_descriptor(*scene_constants);
+
+	for (uint32_t i = 0; i < kDepthCount; ++i)
+	{
+		VkDescriptorImageInfo depth_texture_descriptor = {};
+		depth_texture_descriptor.sampler               = point_sampler;
+		depth_texture_descriptor.imageView             = depth_image_view[(i + 1) % kDepthCount]->get_handle();
+		depth_texture_descriptor.imageLayout           = VK_IMAGE_LAYOUT_DEPTH_STENCIL_READ_ONLY_OPTIMAL;
+
+		std::vector<VkWriteDescriptorSet> write_descriptor_sets = {
+		    vkb::initializers::write_descriptor_set(gather_descriptor_set[i], VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 0, &scene_constants_descriptor),
+		    vkb::initializers::write_descriptor_set(gather_descriptor_set[i], VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, 1, &depth_texture_descriptor),
+		};
+		vkUpdateDescriptorSets(get_device().get_handle(), static_cast<uint32_t>(write_descriptor_sets.size()), write_descriptor_sets.data(), 0, NULL);
+	}
+
+	{
+		std::vector<VkWriteDescriptorSet> write_descriptor_sets;
+		write_descriptor_sets.resize(3);
```

Comment:
You can just write
`		std::vector<VkWriteDescriptorSet> write_descriptor_sets( 3 );
`
to get a vector of size 3.

Reply:
Fixed.